beacon_node/client/src/lib.rs

```diff
@@ -47,7 +49,7 @@ pub struct Client<T: BeaconChainTypes> {
 
 impl<T> Client<T>
 where
-    T: BeaconChainTypes + InitialiseBeaconChain<T> + Clone + 'static,
+    T: BeaconChainTypes + InitialiseBeaconChain<T> + Clone + Send + Sync + 'static,
```

Comment:
As far as I can tell the source of these `Send` and `Sync` bounds is the `NetworkService<T>` being added to the Hyper request extensions. `NetworkService<T>` contains a `PhantomData<T>` which is only `Send + Sync` if `T` is?

We could consider adding `Send + Sync + 'static` as super-traits of `BeaconChainTypes`, which would absolve us of having to repeat them.

On my branch I've removed the `Clone` bound here, as it was caused by gRPC. It would be nice to avoid all erroneous bounds on type-level params, and avoid repeating the ones that are necessary evils.